src/Resolver/AccessResolver.php

```diff
@@ -57,14 +59,16 @@ private static function isMutationRootField(ResolveInfo $info): bool
     /**
      * @return Promise|mixed
      */
-    private function checkAccessForStrictMode(callable $accessChecker, callable $resolveCallback, array $resolveArgs = [])
+    private function checkAccessForStrictMode(callable $accessChecker, callable $resolveCallback, array $resolveArgs = [], bool $nullOnDenied = false)
     {
         $promiseOrHasAccess = $this->hasAccess($accessChecker, $resolveArgs);
-        $callback = function ($hasAccess) use ($resolveArgs, $resolveCallback) {
+        $callback = function ($hasAccess) use ($resolveArgs, $resolveCallback, $nullOnDenied) {
             if (true === $hasAccess) {
```

Comment:
```suggestion
            if ($hasAccess) {
```
No need to check boolean variables with `===` in `if` conditions

Reply:
`true === $hasAccess` is a matter of security we don't want wrong return types (string, object, ...) to give access to a protected field.

Reply:
I thought $hasAccess is `bool`. I would change the naming of this variable, because it's confusing. Or we can allow only boolean values for `hasAccess`

Reply:
Yes, we could enforce it as a boolean I think. 
It's just that if people used to return non boolean value from access callback, their code will break (but if we target v1.0.0, it's ok right ?).
So `hasAccess` should return a boolean or a promise resolving to a boolean. 
Maybe we could check the return type of the callback and the promise and raise an explicit exception if it's not a boolean ?

Reply:
@Vincz I meant the argument `$hasAccess` should be bolean, so that we don't do: `true === $hasAccess`. But anyway, this is not a part of this PR so I just mark this conversation as resolved